xarray/backends/h5netcdf_.py

```diff
@@ -16,6 +16,7 @@
 
 class H5NetCDFArrayWrapper(BaseNetCDF4Array):
     def __getitem__(self, key):
+        key = key.to_tuple() if hasattr(key, 'to_tuple') else key
```

Comment:
> all keys go through Variable.__getitem__ to get here.

I changed `to_tuple` from a method to a function.

I found [some codes](https://github.com/pydata/xarray/blob/master/xarray/conventions.py#L407) call this line without passing through `Variables._broadcast_indexes`.
I'm a bit wondering if it might be a bit error prone.